spec/integration/provider/package_spec.rb

```diff
@@ -12,7 +12,7 @@
         options = {:name => "nosuch#{provider.name}", :provider => provider.name}
 
         pkg = Puppet::Type.newpackage(options)
-        expect { pkg.provider.install }.to raise_error
+        expect { pkg.provider.install }.to raise_error(/source ?(parameter)?|gem|pip|lock file/)
```

Comment:
This test is mildly infuriating :). This `raise_error` captures one of *many* different exceptions, so we can't expect just one. Thus, I've resorted to capturing small fragments of the exception messages in this regex.